test/src/test/kotlin/graphql/nadel/tests/next/NadelIntegrationTest.kt

```diff
@@ -0,0 +1,317 @@
+        val serviceExecutions = services.associate {
+            val serviceName = it.name
+            serviceName to GraphQLServiceExecution(
+                serviceName = serviceName,
+                graphQL = makeServiceGraphQL(it).build(),
+            )
+        }
+
+        return NadelSchemas.newNadelSchemas()
+            .overallSchemas(overallSchemas)
+            .underlyingSchemas(underlyingSchemas)
+            .serviceExecutionFactory { service ->
+                serviceExecutions[service]!!
+            }
+    }
+
+    open fun makeServiceGraphQL(service: Service): GraphQL.Builder {
+        return GraphQL
+            .newGraphQL(
+                SchemaGenerator()
+                    .makeExecutableSchema(
+                        SchemaParser().parse(service.underlyingSchema),
+                        RuntimeWiring.newRuntimeWiring()
+                            .also(service.runtimeWiring)
+                            .build(),
+                    ),
+            )
+            .instrumentation(
+                object : Instrumentation {
+                    override fun instrumentExecutionResult(
+                        executionResult: ExecutionResult,
+                        parameters: InstrumentationExecutionParameters,
+                        state: InstrumentationState?,
+                    ): CompletableFuture<ExecutionResult> {
+                        return CompletableFuture.completedFuture(
+                            executionCapture.capture(
+                                service = service.name,
+                                query = parameters.query,
+                                variables = parameters.variables,
+                                result = executionResult,
+                            ),
+                        )
+                    }
+                },
+            )
+    }
+
+    open fun makeInstrumentation(): NadelInstrumentation {
+        return object : NadelInstrumentation {
+        }
+    }
+
+    open fun getTestData(): TestData {
+        return Class.forName(this::class.qualifiedName + "Data")
+            .getDeclaredConstructor()
+            .newInstance() as TestData
+    }
+
+    open fun assert(result: ExecutionResult) {
+    }
+
+    private fun assertServiceCalls(testData: TestData) {
+        // Unmatched calls, by the end of the function both should be empty if they're matched
+        val unmatchedExpectedCalls = testData.calls.toMutableList()
+        val unmatchedActualCalls = executionCapture.calls.toMutableList()
+
+        unmatchedActualCalls.forEachElementInIterator { iterator, actualCall ->
+            fun getCanonicalQuery(query: String): String {
+                return AstPrinter.printAstCompact(Parser().parseDocument(query))
+            }
+
+            val actualQuery = getCanonicalQuery(actualCall.query)
+            val actualVariables = actualCall.variables
+
+            val matchingCalls = unmatchedExpectedCalls
```

Comment:
It seems like the ordering of the calls in the test data is not relevant. This is true most of the time. For delayed responses the order might vary depending on how long each resolver takes to deliver the data, but it's important that we assert that the very last delayed response is the only one with `hasNext: false`

Reply:
Yeah if you look at `findUnmatchedDelayedResponses` I'm doing that :)

Reply:
That function asserts the overall response's delayed responses though. I still need to match the delayed responses of the underlying service calls.